src/overlays/actors/ovl_En_Weather_Tag/z_en_weather_tag.c

```diff
@@ -127,82 +127,86 @@ void EnWeatherTag_Init(Actor* thisx, GlobalContext* globalCtx) {
-u8 WeatherTag_CheckEnableWeatherEffect(EnWeatherTag* this, GlobalContext* globalCtx, u8 arg2, u8 arg3, u8 arg4, u8 arg5,
-                                       u16 arg6, u8 weatherMode) {
+u8 WeatherTag_CheckEnableWeatherEffect(EnWeatherTag* this, GlobalContext* globalCtx, u8 arg2, u8 arg3, u8 arg4,
+                                       u8 nextLightConfig, u16 arg6, u8 weatherMode) {
     s32 pad;
     u8 ret = false;
     Player* player = GET_PLAYER(globalCtx);
 
     if (Actor_WorldDistXZToActor(&player->actor, &this->actor) < WEATHER_TAG_RANGE100(this->actor.params)) {
-        if ((globalCtx->envCtx.indoors != 0) || !gSkyboxBlendingEnabled ||
-            (globalCtx->skyboxId != SKYBOX_NORMAL_SKY && globalCtx->envCtx.unk_1F == globalCtx->envCtx.unk_20)) {
-            D_8011FB38 = 1;
-            if (globalCtx->envCtx.gloomySkyMode == 0 &&
-                (globalCtx->envCtx.indoors != 0 || (globalCtx->envCtx.unk_1F != 1 && globalCtx->envCtx.unk_21 == 0))) {
-                D_8011FB38 = 0;
+        if (globalCtx->envCtx.indoors || !gSkyboxIsChanging ||
+            (globalCtx->skyboxId != SKYBOX_NORMAL_SKY &&
+             globalCtx->envCtx.lightConfig == globalCtx->envCtx.nextLightConfig)) {
+            gInterruptSongOfStorms = true;
+            if (globalCtx->envCtx.stormRequest == STORM_REQUEST_NONE &&
+                (globalCtx->envCtx.indoors ||
+                 (globalCtx->envCtx.lightConfig != 1 && !globalCtx->envCtx.weatherChgLights))) {
+                gInterruptSongOfStorms = false;
                 if (gWeatherMode != weatherMode) {
                     gWeatherMode = weatherMode;
-                    if (globalCtx->envCtx.gloomySkyMode == 0) {
-                        globalCtx->envCtx.unk_19 = 1;
-                        globalCtx->envCtx.unk_17 = arg2;
-                        globalCtx->envCtx.unk_18 = arg3;
-                        globalCtx->envCtx.unk_1A = arg6;
-                        globalCtx->envCtx.unk_21 = 1;
-                        globalCtx->envCtx.unk_1F = arg4;
-                        globalCtx->envCtx.unk_20 = arg5;
-                        D_8011FB34 = arg5;
-                        globalCtx->envCtx.unk_24 = arg6;
-                        globalCtx->envCtx.unk_22 = globalCtx->envCtx.unk_24;
+                    if (globalCtx->envCtx.stormRequest == STORM_REQUEST_NONE) {
+                        globalCtx->envCtx.weatherChgSkyState = WEATHER_CHANGE_SKY_REQUESTED;
+                        globalCtx->envCtx.skyboxConfig = arg2;
+                        globalCtx->envCtx.nextSkyboxConfig = arg3;
+                        globalCtx->envCtx.weatherChgSkyTimer = arg6;
+                        globalCtx->envCtx.weatherChgLights = true;
+                        globalCtx->envCtx.lightConfig = arg4;
+                        globalCtx->envCtx.nextLightConfig = nextLightConfig;
+                        gSavedNextLightConfig = nextLightConfig;
+                        globalCtx->envCtx.weatherChgDuration = arg6;
```

Comment:
Can probably name the args with this.

Reply:
could, but I also see it as something that could be cleaned up with the actor in general. I may do the rest of them though